### ISSUE TITLE
EmoSens ｖ3.7 and later now complete.

### ISSUE THREAD

**Issue by muooon** [kozistr/pytorch_optimizer#482]
Paper
https://huggingface.co/muooon/EmoNAVI/raw/main/emo-v38-paper(ENG).txt
Paper **"DOI" Acquired Version**  
https://huggingface.co/muooon/EmoTion-Optimizer
Code
https://github.com/muooon/EmoSens

kozistr さん、たびたびの更新でお手数を掛けます。EmoSens 完成しました。
これまでの知見をすべて注ぎ込んだ自信作です。 完全自動学習率を達成しています。恐らくですが、最近知ったばかりなのですが、Grokking というんですかね、それを超えていける、Grokking せずに超えていける、かも？ という実装です(もちろん optimizer 単体で) それとこれをもっと簡単にできないか？というところからも新手法を提案しています、詳しくは論文をご覧ください。 お時間ある時に EmoSens (とバリエーションの４種)を pytorch-optimizer に加えてください。どうぞ宜しくお願いします。

Kozistr, thank you for your patience with the frequent updates. EmoSens is finally complete.

This is my most confident work to date, consolidating everything I've learned into a single system. It achieves a fully automatic learning rate.
Interestingly, I recently came across the concept of "Grokking," and I believe this implementation might actually bypass the need for it entirely—achieving flat minima directly (all within the optimizer itself). We’ve also proposed a new, more streamlined methodology born from the question: "Can't we make this simpler?" Please refer to the paper for the technical details.
When you have a moment, I would be honored if you could add EmoSens (along with its 4 variations, Airy, Cats, and Tion) to pytorch-optimizer. Thank you as always for your support!

**Comment by muooon** [comment]
ちょっと異常性能を観測したので連絡します、｢cita｣欄に、上記のpaperへのリンクを入れて頂けるようにお願いします。
I'm contacting you because I observed some unusual performance. Please add a link to the above paper in the “cita” field.

---

<img width="1149" height="1152" alt="Image" src="https://github.com/user-attachments/assets/ce016e41-448a-40d7-99ee-428284f34ef3" />

emosens：red、emoairy：gray、emocats：blue、
emoairy：smooth：0.0178、value：0.0172、step：661、time：46.85 min
emocats：smooth：0.0157、value：0.0165、step：600、time：40.21 min
emoairy：smooth： 0.0157、value：0.0164、step：600、time：40.61 min
画像１枚、Rank16/Alpha16、全層LoRA、fp8(AMP-bf16)、1batch、600epoch(600step相当) です
Image: 1, Rank16/Alpha16, All layers LoRA, fp8 (AMP-bf16), 1 batch, 600 epochs (equivalent to 600 steps)

過学習傾向：ナシ、焼き付き：ナシ(普通ならコピーになる)、無関係プロンプトへの影響ナシ、普通に使えるLoRAでした。ええ？！ってなりませんか？
Overfitting tendency: None, Burn-in: None (normally would be copied), No impact on unrelated prompts, It was a perfectly usable LoRA. Seriously?! Doesn't that make you go “Huh?!”

とりあえず断言しても良さそうと思えるのは、LR推定がとても優秀。７次moment近似まで活用することで高度なLR予測の正確性で優れている、これは相当な確度としていえる、と思います。
For now, I think it's safe to state that the LR estimation is exceptionally good. By utilizing up to seventh-order moment approximation, it excels in the accuracy of advanced LR predictions—this can be considered a significant degree of precision.

**Comment by muooon** [comment]
[kozistr](https://github.com/kozistr) さん、ご無沙汰しました。この issue は取り下げるべきなのでしょうね close にしようと思います。
もしお時間あれば最新版をご覧ください。 closure 問題を解決し動作する設計になってます。 それと Scheduler も公開しました。 可能であれば Scheduler も新たに登録してくださると嬉しいです。

It’s been a while. I suppose this issue should be withdrawn, so I’m going to mark it as closed.
If you have a moment, please take a look at the latest version. It resolves the closure issue and now features a functional design. I’ve also released the Scheduler. If possible, I’d appreciate it if you could register the Scheduler as a new entry.
